build/flaws/broken-links.js

```diff
@@ -54,39 +79,25 @@ function getBrokenLinksFlaws(doc, $, { rawContent }, level) {
   // us from calling `findMatchesInText()` more than once.
   const matches = new Map();
 
-  function mutateLink($element, suggestion, enUSFallback) {
-    if (suggestion) {
-      $element.attr("href", suggestion);
-    } else if (enUSFallback) {
-      $element.attr("href", enUSFallback);
-      // This functionality here should match what we do inside
-      // the `web.smartLink()` function in kumascript rendering.
-      $element.text(`${$element.text()} (${DEFAULT_LOCALE})`);
-      $element.addClass("only-in-en-us");
-      $element.attr("title", "Currently only available in English (US)");
-    } else {
-      $element.addClass("page-not-created");
-      $element.attr("title", "This is a link to an unwritten page");
-    }
-  }
-
   // A closure function to help making it easier to append flaws
   function addBrokenLink(
     $element,
     index,
     href,
     suggestion = null,
     explanation = null,
-    enUSFallback = null
+    enUSFallback = null,
+    isSelfLink = false
   ) {
+    mutateLink($element, { suggestion, enUSFallback, isSelfLink });
```

Comment:
This here is the most valuable change of all. Now, the `mutateLink` is ALWAYS called from within this function, independent of running in preview mode or running the the `BUILD_FLAWLEVELS="*:ignore, unsafe_html:error"` like we do in prod. 

So, now it reads much better. 

If you look carefully, this is what USED to happen before:

```js
if (reasonWhyThisIsBroken($element.href)) {
  let suggestion = null;
  let redirectsTo = isRedirect($element.href)
  if (redirectsTo) {
    suggestion = redirectsTo
  }
  addBrokenLink($element, suggestion, ...);
}
```
So far so good. But what inside that `addBrokenLink()` function it *used* to do this:
```js
function addBrokenLink($element, suggestion = null, ...)  {
  if (level === FLAW_LEVELS.IGNORE) {
    mutateLink($element)    // <--- HERE WAS THE BUG!
    return;
  }
  let explanation = "Can't resolve $element.href"
  ...
  mutateLink($element, suggestion);
  recordDocumentFlaw($element, explanation, suggestion, ...)
}
```

So that fork depending on `level === FLAW_LEVELS.IGNORE`, too many other things happened afterwards. It accidentally called the `mutateLink()` incorrectly. And we didn't notice because we didn't go down that fork in local development or in our end-to-end tests because they run `yarn build` without setting flaws to ignore.